src/components/dialogs/input-hooks.js

```diff
@@ -289,18 +398,200 @@ export const useConnectivityValue = ({
+                            }) +
+                            ' ' +
+                            (!validation.isFieldRequired
+                                ? intl.formatMessage({
+                                      id: 'Optional',
+                                  })
+                                : '')
+                        }
+                        value={value}
+                    />
+                )}
+            />
+            // </Grid>
+        );
+    }, [
+        label,
+        value,
+        handleChangeValue,
+        formProps,
+        values,
+        intl,
+        validation.isFieldRequired,
+    ]);
+
+    return [value, field];
+};
+
+export const useSearchEquipmentField = ({ handleOpenSearchDialog, label }) => {
+    const classes = useStyles();
+
+    const button = useMemo(() => {
+        return (
+            <Tooltip
+                title={<FormattedMessage id={label} />}
+                placement="top"
+                arrow
+                enterDelay={DELAY}
+                enterNextDelay={DELAY}
+                classes={{ tooltip: classes.tooltip }}
+            >
+                <Button onClick={handleOpenSearchDialog}>
+                    <FindInPageIcon />
+                </Button>
+            </Tooltip>
+        );
+    }, [label, handleOpenSearchDialog, classes.tooltip]);
+    return button;
+};
+
+export const useEnumValue = ({
+    label,
+    defaultValue,
+    validation = {},
+    inputForm,
+    formProps,
+    enumValues,
+}) => {
+    const intl = useIntl();
+    const [value, setValue] = useState(defaultValue);
+
+    useEffect(() => {
+        function validate() {
+            return true;
+        }
+        inputForm.addValidation(label, validate);
+    }, [label, validation, inputForm, value]);
+
+    const handleChangeValue = useCallback((event) => {
+        setValue(event.target.value);
+    }, []);
+
+    const field = useMemo(() => {
+        return (
+            <FormControl fullWidth size="small">
+                {/*This InputLabel is necessary in order to display
+                            the label describing the content of the Select*/}
+                <InputLabel id="enum-type-label" variant={'filled'}>
+                    {intl.formatMessage({
+                        id: label,
+                    }) +
+                        ' ' +
+                        (!validation.isFieldRequired
+                            ? intl.formatMessage({
+                                  id: 'Optional',
+                              })
+                            : '')}
+                </InputLabel>
+                <Select
+                    id={label}
+                    value={value}
+                    onChange={handleChangeValue}
+                    variant="filled"
+                    fullWidth
+                    {...formProps}
+                >
+                    {enumValues.map((e) => (
+                        <MenuItem value={e.id} key={e.id}>
```

Comment:
we are not sure that there is no duplicate id (key will be the same)

Reply:
Fixed by using index